ddl/lightning/env.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lightning
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+
+	"github.com/pingcap/tidb/br/pkg/lightning/log"
+	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/util/logutil"
+	"github.com/pingcap/tidb/util/size"
+	"go.uber.org/zap"
+)
+
+var (
+	// LitBackCtxMgr is the entry for the lightning backfill process.
+	LitBackCtxMgr backendCtxManager
+	// LitMemRoot is used to track the memory usage of the lightning backfill process.
+	LitMemRoot MemRoot
+	// LitDiskRoot is used to track the disk usage of the lightning backfill process.
+	LitDiskRoot DiskRoot
+	// LitRLimit is the max open file number of the lightning backfill process.
+	LitRLimit uint64
+	// LitSortPath is the sort path for the lightning backfill process.
+	LitSortPath string
+	// LitInitialized is the flag indicates whether the lightning backfill process is initialized.
+	LitInitialized bool
+)
+
+// InitGlobalLightningEnv initialize Lightning backfill environment.
+func InitGlobalLightningEnv() {
+	log.SetAppLogger(logutil.BgLogger())
+	sPath, err := genLightningDataDir()
+	if err != nil {
+		logutil.BgLogger().Warn(LitWarnEnvInitFail, zap.Error(err),
+			zap.Bool("lightning is initialized", LitInitialized))
+		return
+	}
+	LitSortPath = sPath
+	maxMemLimit := genMaxMemoryLimit()
+	LitMemRoot = NewMemRootImpl(int64(maxMemLimit), &LitBackCtxMgr)
+	LitDiskRoot = NewDiskRootImpl(LitSortPath, &LitBackCtxMgr)
+	err = LitDiskRoot.UpdateUsageAndQuota()
+	if err != nil {
+		logutil.BgLogger().Warn(LitErrUpdateDiskStats, zap.Error(err),
+			zap.Bool("lightning is initialized", LitInitialized))
+		return
+	}
+	LitBackCtxMgr.init(LitMemRoot, LitDiskRoot)
+	LitRLimit = genRLimit()
+	LitInitialized = true
+	logutil.BgLogger().Info(LitInfoEnvInitSucc,
+		zap.Uint64("memory limitation", maxMemLimit),
+		zap.Uint64("sort path disk quota", LitDiskRoot.MaxQuota()),
+		zap.Uint64("max open file number", LitRLimit),
+		zap.Bool("lightning is initialized", LitInitialized))
+}
+
+func genRLimit() uint64 {
+	rLimit := uint64(1024)
+	var rl syscall.Rlimit
+	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rl)
```

Comment:
linux only?

Reply:
Yes. I think we don't have to consider other platforms.